src/main/java/seedu/address/storage/JsonAdaptedInternshipApplication.java

```diff
@@ -101,7 +115,14 @@ public InternshipApplication toModelType() throws IllegalValueException {
             final Email modelEmail = new Email(contact.get(1));
             final Contact modelContact = new Contact(modelPhone, modelEmail);
 
-            return new InternshipApplication(modelCompanyName, modelJobTitle, modelContact, modelStatus);
+            if (modelInterviewDate != null) {
+                return new InternshipApplication(modelCompanyName, modelJobTitle, modelContact, modelStatus,
+                        modelInterviewDate);
+            } else {
+                return new InternshipApplication(modelCompanyName, modelJobTitle, modelContact, modelStatus);
+            }
```

Comment:
I'm thinking if there's better way of handling null `modelInterviewDate` as I have seen this if-else logic appearing too many times at many places